src/data.jl

```diff
@@ -1186,9 +1178,9 @@ function reconstruct_compound(f::JLDFile, T::String, dt::H5Datatype,
 
     # Now reconstruct the type
     reconname = gensym(T)
-    eval(ReconstructedTypes, Expr(:composite_type, reconname, Core.svec(),
-                                  Core.svec(dt.names...), Any,
-                                  types, false, 0))
+    eval(ReconstructedTypes,
+         Expr(VERSION >= v"0.7.0-DEV.1258" ? :struct_type : :composite_type, reconname,
```

Comment:
this is for Julia PR 23157, right? be sure to run contrib/commit-name.sh on the merge commit
```
$ contrib/commit-name.sh dc819314aa37de4e03652e7b0c4d2af68251bc5e
0.7.0-DEV.1263
```
as there will be master commits higher than 1258 where struct_type won't work